toolchain/sem_ir/type.h

```diff
@@ -207,6 +207,11 @@ class TypeStore : public Yaml::Printable<TypeStore> {
   llvm::SmallVector<TypeId> complete_types_;
 };
 
+// If `type_id` is a `PatternType`, returns its scrutinee type; otherwise,
+// returns `type_id` unchanged.
+auto ExtractScrutineeType(const TypeStore& types, SemIR::TypeId type_id)
```

Comment:
```suggestion
auto ExtractScrutineeType(const File& sem_ir, SemIR::TypeId type_id)
```
Working with a store in a free function is a bit unusual. I think either it should be a method on the store, or it should take a File as its parameter here instead. I'd go with the latter.